indico/modules/events/management/controllers/settings.py

```diff
@@ -136,9 +139,21 @@ def render_form(self, form):
 
 
 class RHEditEventClassification(RHEditEventDataBase):
-    form_class = EventClassificationForm
     section_name = 'classification'
 
+    @property
+    def form_class(self):
+        if allowed_keywords := misc_settings.get('allowed_keywords'):
+            allowed_keywords_set = set(allowed_keywords)
```

Comment:
I think we need to use `set(allowed_keywords) | set(self.event.keywords)` here: When you remove a keyword from the whitelist, events that already have it keep and this removal should not prevent someone from saving this form.

(Of course the keyword would disappear for good once it's removed from the event)

Reply:
Hmmmm my approach was more drastic: only allowed events are displayed, and if saved, only allowed events will be saved, discarding non-allowed ones. It's true this is a side effect, but maintains the database consistency 🤔 

Still, I'll follow your suggestion and allow them all.